src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double RATIO = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] internalArray;
+    private int size = 0;
+
+    public ArrayList() {
+        internalArray = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        size += 1;
+        if (size > internalArray.length) {
+            int newCapacity = getNewCapacity();
+            Object[] newInternalArray = new Object[newCapacity];
+            System.arraycopy(internalArray, 0, newInternalArray, 0, internalArray.length);
+            internalArray = newInternalArray;
+        }
+        internalArray[size - 1] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index not in range");
+        }
+        size += 1;
+        int newCapacity = size > internalArray.length
+                ? getNewCapacity()
+                : internalArray.length;
+        Object[] newInternalArray = new Object[newCapacity];
+        System.arraycopy(internalArray, 0, newInternalArray, 0, index);
+        newInternalArray[index] = value;
+        System.arraycopy(internalArray, index, newInternalArray, index + 1, size - (index + 1));
+        internalArray = newInternalArray;
```

Comment:
```suggestion
        size += 1;
        if (size > internalArray.length) {
            increaseCapacity();
        }
        System.arraycopy(internalArray, index, internalArray, index + 1, size - 1 - index);
        internalArray[index] = value;
```
increase the internalArray before adding value